cg/apps/housekeeper/hk.py

```diff
@@ -350,3 +351,23 @@ def get_files_from_latest_version(
             LOG.info(f"Bundle: {bundle_name} not found in Housekeeper")
             raise HousekeeperBundleVersionMissingError
         return self.files(version=version.id, tags=tags)
+
+    def is_fastq_or_spring_in_all_bundles(self, bundle_names: List[str]) -> bool:
+        """Return bool of fastq or spring present in all bundles."""
+        sequencing_files_in_hk: Dict[str, bool] = {}
+        for bundle_name in bundle_names:
+            sequencing_files_in_hk[bundle_name] = False
+            for tag in [SequencingFileTag.FASTQ, SequencingFileTag.SPRING_METADATA]:
+                sample_file_in_hk: List[bool] = []
+                hk_files: Optional[List[File]] = self.get_files_from_latest_version(
+                    bundle_name=bundle_name, tags=[tag]
+                )
+                try:
+                    sample_file_in_hk += [True for hk_file in hk_files if hk_file.is_included]
+                    if sample_file_in_hk:
+                        break
+                finally:
+                    sequencing_files_in_hk[bundle_name] = (
+                        all(sample_file_in_hk) if sample_file_in_hk else False
+                    )
+        return all(sequencing_files_in_hk.values())
```

Comment:
I have struggled a bit with this one.

I added the try-finally because I think that python will throw an error if you try to iterate over a NoneType, but I reallize now that the worst case scenario will be an empty list which is handled gracefully by python and then we should not have to use the try clause.

I will remove the try part. 👍 

The break is because if we have fastq and not spring in the same bundle (or vice versa), once `sample_file_in_hk` is not empty, we will add `False` to the array (for the other missing file type) and we will incorrectly not be able to remove it (assuming that if one file type is included then both are; independent of present or not).